src/kbmod/image_info.py

```diff
@@ -12,15 +17,33 @@ def __init__(self):
         self.wcs = None
         self.center = None
         self.obs_code = ""
+        self.filename = None
+        self.visit_id = None
 
-    def populate_from_fits_file(self, filename):
+    def populate_from_fits_file(self, filename, visit_in_filename=None):
         """Read the file stats information from a FITS file.
 
         Parameters
         ----------
         filename : string
             The path and name of the FITS file.
+        visit_in_filename : list of ints
+            A list containg the first last character of the visit ID
+            contained in the filename and the first character after the visit ID
+            (e.g. [0, 6] will use characters from 0 to 5 inclusive).
         """
+        # Skip non-FITs files.
+        if len(filename) < 5 or filename[-5:] != ".fits":
+            return
+
+        # If visit_in_filename is provided extract the visit_id using that
+        # otherwise use the filename minus the suffix.
+        visit_name = filename.rsplit("/")[-1]
+        if visit_in_filename is not None and len(visit_name) > visit_in_filename[1] + 5:
+            self.visit_id = visit_name[visit_in_filename[0] : visit_in_filename[1]]
+        else:
+            self.visit_id = visit_name[:-5]
```

Comment:
We should also probably extract the `visit_id` from the FITS file itself, if we area already opening it anyhow. From what I can see it's under `IDNUM`  key. There is also an `EXPID` key, which I think combines the `IDNUM` and `CCDNUM`.

Reply:
Is this always the case? Or is that just for dec cam data? I'm always wary about how much we can truly extract from FITS files.